Sources/TuistMigration/Utilities/TargetsExtractor.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+import PathKit
+import TSCBasic
+import TuistSupport
+import XcodeProj
+
+/// An interface to extract all targets from an xcode project, sorted by number of dependencies
+public protocol TargetsExtracting {
+    /// - Parameters:
+    ///   - xcodeprojPath: Path to the Xcode project.
+    func targetsSortedByDependencies(xcodeprojPath: AbsolutePath) throws -> [(targetName: String, dependenciesCount: Int)]
+}
+
+public final class TargetsExtractor: TargetsExtracting {
+    // MARK: - Init
+
+    public init() {}
+
+    // MARK: - EmptyBuildSettingsChecking
+
+    public func targetsSortedByDependencies(xcodeprojPath: AbsolutePath) throws -> [(targetName: String, dependenciesCount: Int)] {
+        guard FileHandler.shared.exists(xcodeprojPath) else { throw TargetsExtractorError.missingXcodeProj(xcodeprojPath) }
+        let pbxproj = try XcodeProj(path: Path(xcodeprojPath.pathString)).pbxproj
+        let targets = pbxproj.nativeTargets + pbxproj.aggregateTargets + pbxproj.legacyTargets
+        if targets.isEmpty {
+            throw TargetsExtractorError.noTargets
+        }
+        do {
+            return try sortTargetsByDependenciesCount(targets)
+        } catch {
+            throw TargetsExtractorError.failedToExtractTargets(error.localizedDescription)
```

Comment:
I don't think we need to map the error into another error. We can remove this entire `do/catch` and bubble up the error that we are getting from `XcodeProj.